tests/gradients/test_gradient_transform.py

```diff
@@ -395,15 +395,15 @@ def test_shots_error(self):
         """Raise an exception if shots is used within the QNode"""
         dev = qml.device("default.qubit", wires=1, shots=1000)
 
-        @qml.qnode(dev)
         def circuit(x, shots):
             qml.RX(x, wires=0)
             return qml.expval(qml.PauliZ(0))
 
-        with pytest.raises(
-            ValueError, match="'shots' argument name is reserved for overriding the number of shots"
-        ):
-            qml.gradients.param_shift(circuit)(0.2, shots=100)
+        with pytest.warns(UserWarning, match="Detected 'shots' as an argument to the given"):

```

Comment:
I'm confused: how come this was changed and the warning message refers to the `shots` argument?

Reply:
This is an unrelated warning that I noticed was being thrown by this test. Basically, initializing a QNode with a function that has an argument named 'shots' will always throw a warning about it. At the same time, we also raise an error whenever a gradient transform is applied to quantum functions with a 'shots' argument (the reason being that the user argument conflicts with PennyLane's use of the shots argument).

The test was only catching the error but not the warning, so this change remedies this.

Reply:
Awesome, good catch! Thanks for the explanation.